src/main/kotlin/no/nav/medlemskap/services/ereg/EregClient.kt

```diff
@@ -0,0 +1,45 @@
+package no.nav.medlemskap.services.ereg
+
+import io.github.resilience4j.retry.Retry
+import io.ktor.client.features.ClientRequestException
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.client.request.url
+import io.ktor.http.ContentType
+import io.ktor.http.HttpHeaders
+import no.nav.medlemskap.common.cioHttpClient
+import no.nav.medlemskap.services.runWithRetryAndMetrics
+
+class EregClient (
+        private val baseUrl: String,
+        private val retry: Retry? = null
+) {
+    suspend fun hentEnhetstype(orgnummer:String, callId: String, consumerId: String): String? {
+        val organisasjon = runCatching {
+            runWithRetryAndMetrics("Ereg", "noekkelinfo", retry) {
+                cioHttpClient.get<Organisasjon> {
+                    url("$baseUrl/v1/organisasjon/$orgnummer/noekkelinfo")
+                    header(HttpHeaders.Accept, ContentType.Application.Json)
+                    header("Nav-Call-Id", callId)
+                    header("Nav-Consumer-Token", consumerId)
```

Comment:
Forventer de et token eller en id her..? 
I f.eks MedlClient, så sender vi inn en consumerId, som er ment å identifisere de kallende systemet, men ikke brukes til autentisering. (header("Nav-Consumer-Id", configuration.sts.username)) I AaRegClient sender vi inn en consumer token, og de er da et oidc-token (header("Nav-Consumer-Token", "Bearer ${oidcToken}")) Jeg trodde Ereg ikke krevde autentisering, men du har kalt headeren Consumer-Token..?

Reply:
Det er nok en feiltagelse på min side. I testen for eregclienten så kopierte jeg over de som ble gjort i aareg testene, og fikk feilmeldinger om at det ikke var noen consumer-token header til stede, så du har rett her